steps/unable-to-enter/second-failed-attempt.tsx

```diff
@@ -0,0 +1,221 @@
+import formatDate from "date-fns/format";
+import { Heading } from "lbh-frontend-react";
+import {
+  FieldsetLegend,
+  HeadingLevels,
+  LabelProps,
+  List,
+  PageAnnouncement,
+  Textarea,
+} from "lbh-frontend-react/components";
+import { useRouter } from "next/router";
+import React from "react";
+import {
+  ComponentDatabaseMap,
+  ComponentWrapper,
+  DynamicComponent,
+  makeDynamic,
+  StaticComponent,
+} from "remultiform/component-wrapper";
+import { TransactionMode } from "remultiform/database";
+import { Checkboxes, CheckboxesProps } from "../../components/Checkboxes";
+import { makeSubmit } from "../../components/makeSubmit";
+import failedAttemptReasonCheckboxes from "../../helpers/failedAttemptReasonCheckboxes";
+import getProcessRef from "../../helpers/getProcessRef";
+import keyFromSlug from "../../helpers/keyFromSlug";
+import useDataValue from "../../helpers/useDataValue";
+import PageSlugs from "../../steps/PageSlugs";
+import PageTitles from "../../steps/PageTitles";
+import ProcessDatabaseSchema from "../../storage/ProcessDatabaseSchema";
+import Storage from "../../storage/Storage";
+
+const storeSecondVisitDate = async (): Promise<void> => {
+  const db = await Storage.ProcessContext?.database;
+
+  if (!db) {
+    console.error("Unable to find process database");
+
+    return;
+  }
+
+  const processRef = keyFromSlug()();
+  const date = new Date().toISOString();
+
+  await db.transaction(
+    ["unableToEnter"],
+    async (stores) => {
+      const unableToEnter =
+        (await stores.unableToEnter.get(processRef)) ||
+        ({} as ProcessDatabaseSchema["schema"]["unableToEnter"]["value"]);
+
+      await stores.unableToEnter.put(processRef, {
+        ...unableToEnter,
+        secondFailedAttempt: {
+          ...unableToEnter.secondFailedAttempt,
```

Comment:
This method looks identical to the first failed attempt's version, except for the key to use. Let's refactor it into a helper.